crates/test/mock-consensus/src/builder.rs

```diff
@@ -26,9 +30,9 @@ impl Builder {
         self
     }
 
-    pub fn app_state(self, _: ()) -> Self {
-        // this does not do anything yet
-        self
+    pub fn app_state(self, app_state: AppState) -> Self {
```

Comment:
This should be taking a Vec<u8>, so that the mock engine code doesn't build in Penumbra dependencies. Instead, we'll want to have a penumbra_app_state method in an extension trait like we sketched while pairing that does the serialization and calls this method with the resulting bytes.

Reply:
sure!

to explain why this didn't include that trait originally, i had been hoping to add the extension trait in a distinct PR following this one. i'll go ahead and add it now, to preserve that agnosticism.

i also added a note about this as a design principle to #3588, to write that down somewhere. i'll mention it in some crate-level documentation once we're a little further along into porting tests.

see: 17cfd12

**NB:** using `Into<Bytes>` as a parameter ended up panning out a little more nicely, the commit message there goes into a little more detail.

Reply:
Awesome, all good, it just seemed important to get the split in from the beginning rather than trying to tease it apart afterwards

Reply:
agreed, it is difficult to undo that sort of coupling after the fact.

marking as resolved!